edi_oca/models/edi_configuration.py

```diff
@@ -0,0 +1,222 @@
+            "dateutil": safe_eval.dateutil,
+            "time": safe_eval.time,
+            "utc_now": fields.Datetime.now(),
+            "date_to_string": self._date_to_string,
+            "datetime_to_string": self._datetime_to_string,
+            "time_to_string": lambda dt: dt.strftime("%H:%M:%S") if dt else "",
+            "first_of": fields.first,
+        }
+
+    def _get_code_snippet_eval_context(self):
+        """Prepare the context used when evaluating python code
+
+        :returns: dict -- evaluation context given to safe_eval
+        """
+        ctx = {
+            "uid": self.env.uid,
+            "user": self.env.user,
+            "DotDict": DotDict,
+            "conf": self,
+        }
+        ctx.update(self._time_utils())
+        return ctx
+
+    def _evaluate_code_snippet(self, snippet, **render_values):
+        if not self._code_snippet_valued(snippet):
+            return {}
+        eval_ctx = dict(render_values, **self._get_code_snippet_eval_context())
+        safe_eval.safe_eval(snippet, eval_ctx, mode="exec", nocopy=True)
+        result = eval_ctx.get("result", {})
+        if not isinstance(result, dict):
+            return {}
+        return result
+
+    def edi_exec_snippet_before_do(self, record, **kwargs):
+        self.ensure_one()
+        vals = {
+            "todo": kwargs.get("todo", True),
+        }
+        # Execute snippet before do
+        vals_before_do = self._evaluate_code_snippet(
+            self.snippet_before_do, record=record, **kwargs
+        )
+
+        # Prepare data
+        vals.update(
+            {
+                "todo": vals_before_do.get("todo", True),
+                "snippet_do_vars": vals_before_do.get("snippet_do_vars", False),
+                "event_only": vals_before_do.get("event_only", False),
+                "tracked_fields": vals_before_do.get("tracked_fields", False),
+                "edi_action": vals_before_do.get("edi_action", False),
+            }
+        )
+        return vals
+
+    def edi_exec_snippet_do(self, record, **kwargs):
+        self.ensure_one()
+
+        old_value = kwargs.get("old_vals", {}).get(record.id, {})
+        new_value = kwargs.get("vals", {}).get(record.id, {})
+        vals = {
+            "todo": kwargs.get("todo", True),
+            "record": record,
+            "operation": kwargs.get("operation", False),
+            "edi_action": kwargs.get("edi_action", False),
+            "old_value": old_value,
+            "vals": new_value,
+        }
+        if self.snippet_before_do:
+            before_do_vals = self.edi_exec_snippet_before_do(record, **kwargs)
+            vals.update(before_do_vals)
+        if vals["todo"]:
+            return self._evaluate_code_snippet(self.snippet_do, **vals)
+        return True
+
+    @api.model
+    def edi_get_conf(self, trigger, model_name=None, partners=None, backend=None):
+        domain = [("trigger", "=", trigger)]
+        if model_name:
```

Comment:
Just accept the record and retrieve its name from it.
In fact, you should filter only the configurations that are assigned to the current partner.

I re-write here what has been done for PO to give you an example. 
```
    def on_button_confirm_purchase_order(self, records):
        trigger = "on_button_confirm_purchase_order"
        confs = record.edi_purchase_conf_ids.edi_get_conf(trigger)
        for conf in confs:
             conf.edi_exec_snippet_do(record)
```
You see? You always have the recordset of configurations set on the partner. You just have to filter them by trigger.
In fact, this is what I described in the specs :wink: 
And the docstring I provided is "filter current recordset of conf using trigger".